src/kinetics/LinearBurkeRate.cpp

```diff
@@ -0,0 +1,356 @@
+    else {
+        throw InputFileError("LinearBurkeRate::setParameters", m_input,
+            "Collider 'M' for reaction '{}' must be specified in a"
+            " pressure-dependent-Arrhenius (PLOG), falloff (Troe form), or Chebyshev"
+            " format.", eqn);
+    }
+    string eig_eps_key;
+    AnyMap params;
+    params["A"] = 1.0;
+    params["b"] = 0.0;
+    params["Ea"] = 0.0;
+    m_epsObj_M = ArrheniusRate(AnyValue(params), colliders[0].units(), eps_units);
+    m_colliderInfo["M"] = colliders[0];
+    // Start at 1 because index 0 is for "M"
+    for (size_t i = 1; i < colliders.size(); i++){
+        if (!colliders[i].hasKey("name")) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "The collider located at index '{}' in 'colliders' of reaction '{}'"
+                " has no 'name' defined.", i, eqn);
+        }
+        auto nm = colliders[i]["name"].asString();
+        if (!colliders[i].hasKey("efficiency") && !colliders[0].hasKey("eig0")) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "Collider '{}' in reaction '{}' lacks an 'efficiency' or 'eig0' key.",
+                nm, eqn);
+        }
+        if (eig_eps_key.empty()) {
+            if (colliders[i].hasKey("eig0") && !colliders[i].hasKey("efficiency")) {
+                eig_eps_key = "eig0";
+            } else if (colliders[i].hasKey("efficiency") && !colliders[i].hasKey("eig0")) {
+                eig_eps_key = "efficiency";
+            }
+            else{
+                throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                    "Collider '{}' in reaction '{}' cannot contain both 'efficiency'"
+                    " and 'eig0'. Any additional colliders must also make same choice"
+                    " as that of M.", nm, eqn);
+            }
+        } else if (!colliders[i].hasKey(eig_eps_key)) {
+            throw InputFileError("LinearBurkeRate::setParameters", m_input,
+                "All collision efficiencies in reaction '{}' must be defined uniformly"
+                " as either 'eig0' or 'efficiency'. No mixing and matching is allowed.",
+                eqn);
+        }
+        // Save data to m_colliderInfo, which will make it accessible by getParameters
+        m_colliderInfo[colliders[i]["name"].asString()] = colliders[i];
+        m_colliderNames.push_back(colliders[i]["name"].as<string>());
+
+        ArrheniusRate epsObj_i;
+        // eig0 and eps are ONLY interchangeable due to the requirement that eps_M have
+        // parameters {A: 1, b: 0, Ea: 0}
+        params["A"] = colliders[i][eig_eps_key]["A"].asDouble() /
+            colliders[0][eig_eps_key]["A"].asDouble();
+        params["b"] = colliders[i][eig_eps_key]["b"].asDouble() -
+            colliders[0][eig_eps_key]["b"].asDouble();
+        params["Ea"] = colliders[i][eig_eps_key]["Ea"].asDouble() -
+            colliders[0][eig_eps_key]["Ea"].asDouble();
```

Comment:
The current cause of test failures is coming from here, because `colliders[0]` (M) doesn't have a field corresponding to `eig_eps_key`. I think you need to put this normalization of the parameters inside an `if (eig_eps_key == "eig0")` condition, with the alternative being that you can just set `params = colliders[i][eig_eps_key]` as no normalization is nescessary when the parameters are already specified as efficiencies.

Reply:
Do you think it would be confusing for the user that if eig0 is used, then they must specify a value for M, but if efficiency is used, then they do not? I can update the YAML implementation guide on Monday with a note about this

Reply:
I don't even think the formulation is necessarily that restrictive. Since for the underlying calculation you always convert to the efficiency, all that's really required is that every non-M species has either an `eig0` or an `efficiency`, and that if any species has an `eig0`, then you also need an `eig0` for `M`. I don't think it's confusing to skip writing an efficiency for `M`, any more than it is in the case of the other formulations for three-body/falloff reactions.